### PR TITLE
python: remove unneeded name attributes, remove rec, make afew an application

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
As I found out, it's not necessary to specify the `name` for python packages if we set `pname` and `version`. Additionally, get rid of the `rec`, and change the `afew` package to be a python application (what it is, it's not a library).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/curtsies/default.nix [hunk 1]
```diff
@@ -1,10 +1,12 @@
 { stdenv, buildPythonPackage, fetchPypi, blessings, mock, nose, pyte, pytest, wcwidth }:
 
-buildPythonPackage rec {
+let
```
Comment (dezgeg) [inline pkgs/development/python-modules/curtsies/default.nix]: Why the separate let block?

**Comment by flokli** [comment]
to avoid having to use rec in the set below

File: pkgs/applications/networking/mailreaders/afew/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+pythonPackages.buildPythonApplication {
+  inherit pname version;
+
+  src = pythonPackages.fetchPypi {
+    inherit pname version;
+    sha256 = "121w7bd53xyibllxxbfykjj76n81kn1vgjqd22izyh67y8qyyk5r";
+  };
+
+  propagatedBuildInputs = with pythonPackages; [
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/afew/default.nix]: Setuptools_scm is a build time input

File: pkgs/applications/networking/mailreaders/afew/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    setuptools_scm pythonPackages.notmuch chardet
+  ] ++ stdenv.lib.optional (!pythonPackages.isPy3k) subprocess32;
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = "${version}";
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/afew/default.nix]: preBuild is this likely a better place for this one

File: pkgs/applications/networking/mailreaders/afew/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  postInstall = ''
+    wrapProgram $out/bin/afew \
+      --prefix LD_LIBRARY_PATH : ${notmuch}/lib
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/afew/default.nix]: Can this be fixed by hardcoring the path to the lib. Using LD_LIBRARY_PATH should be prevented

File: pkgs/applications/networking/mailreaders/afew/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (FRidh) [inline pkgs/applications/networking/mailreaders/afew/default.nix]: License?

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
Why do you want to prevent the recursive set?

**Comment by flokli** [comment]
@FRidh I was under the impression `rec` is costly and should be avoided if not really needed. Seems I'm mistaken?

Thanks for reviewing the other things, seems like this slipped through in previous PRs. WIll update soonish.

File: pkgs/applications/networking/mailreaders/afew/default.nix (hunk 3 above)
Comment (flokli) [reply pkgs/applications/networking/mailreaders/afew/default.nix]: seems like this isn't required anymore at all, as we're fetching from pypi - and setuptools_scm is able to figure out the version on its own

File: pkgs/applications/networking/mailreaders/afew/default.nix (hunk 4 above)
Comment (flokli) [reply pkgs/applications/networking/mailreaders/afew/default.nix]: It's already hardcoded into `pythonPackages.notmuch`. Will strip this.

**Comment by FRidh** [comment]
There is no need to avoid `rec` here.

**Comment by flokli** [comment]
@FRidh Updated.

**Comment by FRidh** [comment]
@flokli can you squash the `afew` changes.

**Comment by FRidh** [comment]
Included changes in https://github.com/NixOS/nixpkgs/pull/34077.

**Pull request closed without merging**